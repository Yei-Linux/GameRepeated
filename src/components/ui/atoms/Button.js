import React from "react";
import classNames from "classnames";

const Button = ({
  variation = "primary",
  children,
  onClick,
  disable = false,
  type = "button",
  css,
  className,
}) => {
  return (
    <button
      type={type}
      style={css}
      className={classNames(
        className,
        "button",
        `button--${variation}`,
        "button--pd1",
        {
          "button--disable": disable,
        }
      )}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
