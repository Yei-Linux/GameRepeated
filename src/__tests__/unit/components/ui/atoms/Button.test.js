/* eslint-disable testing-library/no-debugging-utils */
import Button from '../../../../../components/ui/atoms/Button'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('The Button component', () => {
  const defaultProps = {
    variation: 'primary',
    children: 'UI Button',
    onClick: jest.fn(),
    disable: false,
    type: 'button',
    css: {},
    className: 'button',
  }

  const setupComponent = (props = defaultProps) => render(<Button {...props} />)

  it('Should render component the user will see', () => {
    const { asFragment } = setupComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('Should style button by variation prop', () => {
    const { children } = defaultProps
    const variation = 'bdprimary'
    setupComponent({
      ...defaultProps,
      variation,
    })

    const button = screen.getByText(children)

    expect(button).toHaveClass(`button--${variation}`)
  })

  it('Should call the click method when it is provided', () => {
    const { children, onClick } = defaultProps
    setupComponent()
    const button = screen.getByText(children)

    userEvent.click(button)

    expect(onClick).toHaveBeenCalledTimes(1)
  })

  it('Should css prop inherit into component', () => {
    const { children } = defaultProps
    const css = {
      width: '400px',
      height: '150px',
    }
    setupComponent({
      ...defaultProps,
      css,
    })
    const button = screen.getByText(children)

    expect(button).toHaveStyle(css)
  })

  it('Should disable button when prop its provided', () => {
    const { children } = defaultProps
    setupComponent({
      ...defaultProps,
      disable: true,
    })

    const button = screen.getByText(children)

    expect(button).toHaveClass('button--disable')
  })
})
