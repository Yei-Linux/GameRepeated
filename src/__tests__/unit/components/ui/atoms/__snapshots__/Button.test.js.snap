// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component Should render component the user will see 1`] = `
<DocumentFragment>
  <button
    class="button button button--primary button--pd1"
    type="button"
  >
    UI Button
  </button>
</DocumentFragment>
`;
